test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -238,17 +263,23 @@ func createVsphereIdentitySecret(ctx context.Context, bootstrapClusterProxy fram
 		})).To(Succeed())
 }
 
-func checkClusterIdentitySecretOwnerRef(ctx context.Context, c ctrlclient.Client) {
+func checkClusterIdentitySecretOwnerRefAndAnnotation(ctx context.Context, c ctrlclient.Client) {
 	s := &corev1.Secret{}
 
```

Comment:
Should we also add some log output here to differentiate between inital and later?